src/main/java/com/scilus/nf/test/nifti/Methods.java

```diff
@@ -0,0 +1,45 @@
+package com.scilus.nf.test.nifti;
+
+import com.ericbarnhill.niftijio.*;
+import com.ericbarnhill.niftijio.tools.*;
+import java.io.IOException;
+import java.util.*;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.nio.charset.StandardCharsets;
+
+public class Methods
+{
+    public static String compare(String i_file) throws IOException
+    {
+
+        NiftiVolume v = NiftiVolume.read(i_file);
+        Nifti1Header f1 = v.getHeader1();
+        NDimensionalArray d = v.getData();
+
+        ArrayList<int[]> indcs = new IndexIterator().iterateReverse(d.getDims());
+
+        String md5 = "";
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            for (int[] indc: indcs) {
+                md.update(String.valueOf(d.get(indc)).getBytes("UTF-8"));
+            }
```

Comment:
Here, instead of reading from the array, I think we can get a `buffer` and feed all the bytes to `md.update`. It would be more efficient, but would require a bit of coding in `niftijio` itself to have it clean. I can get you that in an hour doing only refactoring of the already available code. Tell me !

Reply:
Let's leave this like it is :wink: